src/App.tsx

```diff
@@ -15,36 +19,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETICAL = 'alphabetical',
+  LENGTH = 'length',
+  DEFAULT = 'default',
+}
+
+interface FilterParams {
+  sortType: SortType;
+  reversed: boolean;
+}
+
+const getPreparedGoods = (
+  goods: Good[],
+  { sortType, reversed }: FilterParams,
+): Good[] => {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABETICAL:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [reversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.DEFAULT);
+
+  const handleSortAlphabetically = () => setSortType(SortType.ALPHABETICAL);
+  const handleSortByLength = () => setSortType(SortType.LENGTH);
```

Comment:
Similarly to the previous comment, the function's name should start with a verb. Consider renaming `handleSortByLength` to `sortByLength` or `applyLengthSorting`.